fabfile.py

```diff
@@ -1785,51 +1784,44 @@ def install_elasticsearch():
+                       'vm.max_map_count=262144', True)
         else:
             print(red("Unknown distribution"))
 
-    tmp_local_extract_path = join(local('pwd'), 'estmp')
-    local('mkdir -p %s' % tmp_local_extract_path)
-
-    elasticsearch_exists = False
-    extract_path = normpath(
-        join(env.projectpath, 'var', 'elasticsearch'))
+    base_extract_path = normpath(
+        join(env.projectpath, 'var'))
+    extract_path = join(base_extract_path, 'elasticsearch')
     if exists(extract_path):
-        elasticsearch_exists = True
         print("elasticsearch already installed")
-
-    tar_filename, headers = urlretrieve('https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-{version}.tar.gz'.format(version=ELASTICSEARCH_VERSION))
-    sha1_filename, headers = urlretrieve('https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-{version}.tar.gz.sha1'.format(version=ELASTICSEARCH_VERSION))
-    with open(tar_filename, 'rb') as tar_file:
-        with open(sha1_filename, 'rb') as sha1_file:
-            if sha1(tar_file.read()).hexdigest() != sha1_file.read().strip():
-                print(red("sha1sum of elasticsearch tarball doesn't match, exiting"))
-                sys.exit(1)
-
-        tar_file.seek(0)
-        with tarfile.open(mode='r:gz', fileobj=tar_file) as tar:
-            tar.extractall(path=tmp_local_extract_path)
-
-        # rename var/elasticsearch/elasticsearch-5.2.0 to var/elasticsearch
-        move(join(tmp_local_extract_path, 'elasticsearch-{version}'.format(version=ELASTICSEARCH_VERSION)),
-             join(tmp_local_extract_path, 'elasticsearch'))
-        if elasticsearch_exists:
-            run('rm -rf %s' % extract_path)
-        store_path = join(env.projectpath, 'var')
-        result = put(join(tmp_local_extract_path, 'elasticsearch'), store_path)
+        run('rm -rf %s' % extract_path)
+
+    base_filename = 'elasticsearch-{version}'.format(version=ELASTICSEARCH_VERSION)
+    tar_filename = base_filename + '.tar.gz'
+    sha1_filename = tar_filename + '.sha1'
+    with cd(base_extract_path):
+        if not exists(tar_filename):
+            run('curl -o {fname} https://artifacts.elastic.co/downloads/elasticsearch/{fname}'.format(fname=tar_filename))
+        sha1_expected = run('curl https://artifacts.elastic.co/downloads/elasticsearch/' + sha1_filename)
+        sha1_effective = run('openssl sha1 ' + tar_filename)
+        if ' ' in sha1_effective:
+            sha1_effective = sha1_effective.split(' ')[-1]
+        assert sha1_effective == sha1_expected, "sha1sum of elasticsearch tarball doesn't match, exiting"
+        run('tar zxf ' + tar_filename)
+        run('rm ' + tar_filename)
+        run('mv %s elasticsearch' % base_filename)
 
         # ensure that the folder being scp'ed to belongs to the user/group
-        sudo('chown -R {user}:{group} {path}'.format(
+        run('chown -R {user}:{group} {path}'.format(
```

Comment:
Ah, this may work on our server configurations, but fails on our local setups (especially on mac). Must add a `_group` key to rc files, and default usuage to be `_user`, override as needed.

Reply:
good point. Added env.group (still in single-commit form.)